### PR TITLE
Updated to node20 and latest dependencies

### LINKED ISSUE #58: Node 20/ latest dependencies and workflows

### PR DESCRIPTION
**Pull request by jpolavar**
Closes #58

### CODE REVIEW

**Review by carlansley — APPROVED** [review]

**Comment by carlansley** [comment]
lgtm, i updated the branch protection rules.  Unfortunately you might have to do another small commit to fire off the codeQL actions.

File: package.json [hunk 1]
```diff
@@ -32,34 +31,34 @@
...
     "eslint-plugin-no-secrets": "^0.8.9",
     "eslint-plugin-node": "^11.1.0",
-    "eslint-plugin-sonarjs": "^0.23.0"
+    "eslint-plugin-sonarjs": "^0.24.0"
   },
   "scripts": {
-    "prepublishOnly": "npm run build:dist-types && npm run build:dist-cjs && npm run build:dist-mjs",
+    "prepublishOnly": "npm run build:dist-types && npm run build:dist-mjs",
     "build:dist-types": "rimraf dist-types && npx builder --type=types --outDir=dist-types",
-    "build:dist-cjs": "rimraf dist-cjs && npx builder --type=commonjs --sourceMap --entryPoint=index.ts --outDir=dist-cjs --outFile=index.cjs && echo \"module.exports = module.exports.default;\" >> dist-cjs/index.cjs && node dist-cjs/index.cjs",
```
Comment (carlansley) [inline package.json:53]: one thing I just noticed - you might want to check if eslint works with a pure ESM plugin.  maybe just try the beta out with eslint-config first before squerging this?

**Review by carlansley — CHANGES_REQUESTED** [review]
lgtm, just putting it in "request changes" state until its verified it works with eslint-config

File: package.json (hunk 1 above)
Comment (jpolavar) [reply package.json:53]: Should we convert eslint-config to use esm to test eslint-plugin since there will be discrepancies loading modules

**Comment by jpolavar** [comment]
> lgtm, just putting it in "request changes" state until its verified it works with eslint-config

tested with eslint beta on eslint-config.

Screenshot 2024-03-22 at 1 50 18 PM

**Review by carlansley — APPROVED** [review]

**Pull request merged**